### DISCUSSION TITLE
How to update translation files without a qmake (*.pro) file

### DISCUSSION THREAD

**Discussion by mgrojo** [sqlitebrowser/sqlitebrowser#3451] [Q&A]
Our [Release process](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Release-process) page says to run the following to update the translation files:

```
lupdate src/src.pro
```

But we already got rid of the qmake file `src/src.pro` and the [Qt5 documentation](https://doc.qt.io/qt-5/linguist-manager.html) does not talk about an equivalent procedure using CMake.

I did it in the manual way:

```
lupdate src/*.cpp src/*.ui src/*.h -ts src/translations/*.ts
```

And it seems to work fine. But I'd like to be sure this has not removed something or added something without being needed.

Does someone know a better approach?

**Comment by justinclift** [comment]
Oh, that's a really good question.  Um, no idea for the answer though.  The manual `lupdate` method seems like the right concept. :smile:

**Comment by lucydodo** [comment]
Well, I'm not sure if it's a better approach, but the I used to create a `.pro` file with the `qmake -project` command and work on it.

**Reply by justinclift** [reply]
Clever approach. :smile:

**Comment by scottfurry** [comment]
See [Qt Linguist Manual page](https://doc.qt.io/qt-5/linguist-id-based-i18n.html#advanced-usage) talking about advanced usage of `lupdate` and `lrelease`.

**Reply by scottfurry** [reply]
Thinking about this a little more...
What if (Danger! Will Robinsion):
- add a CMake target of "Translations"
- Target calls a for loop to cycle through all the translations (i.e. en_GB, fr_FR, de_DE, et al)
- each loop iteration executes a couple of CMake commands
- first [`execute_process`](https://cmake.org/cmake/help/latest/command/execute_process.html) does:
  `lupdate -recursive <project-dir> -ts <project-dir>/i18n/myapp-text_($LANG_VARIABLE).ts`
- second `execute_process` follows up with:
  `lrelease -idbased <project-dir>/i18n/myapp-text_($LANG_VARIABLE).ts`
- after the for loop finsishes, all the `.pm` files get scooped up for build

I haven't played with custom CMake targets, but this seems _logical_ to me.

**Reply by mgrojo** [reply]
Qt6 has [support for Qt Linguist in CMake](https://doc.qt.io/qt-6/cmake-commands-qtlinguisttools.html), so for the future, that should be the way to follow. But for now, I think I will compare my approach with @lucydodo's and see if something is missing or shouldn't be there.

**Comment by mgrojo** [comment]
After reviewing thoroughly the changes, it seems this is doing what it has to do:
```
lupdate src/*.cpp src/*.ui src/*.h -ts src/translations/*.ts
```
The other approach is including a lot more strings from other sources, which we didn't have before.
